configure.ac

```
@@ -2187,7 +2187,7 @@ AS_CASE([$target],
     AC_CHECK_FUNC([socketpair], [AC_DEFINE([HAS_SOCKETPAIR], [1])])],
   [*-pc-windows],
     [cclibs="$cclibs ws2_32.lib"
-    AC_SEARCH_LIBS([socket], [ws2_32])
+    OCAML_CL_SEARCH_LIBS([socket], [ws2_32])
```

You know I of course admire the m4sh... but, especially given that one of these days #10505 should eliminate of this, I think this might be tripping into the realm of over-solving it:
```suggestion
    dnl AC_SEARCH_LIBS doesn't work with MSVC; manually update LIBS
    AS_ECHO(["checking for library containing socket... (cached) ws2_32.lib"])
    LIBS="ws2_32.lib $LIBS"
```
?

---

FWIW `OCAML_CL_SEARCH_LIBS` is exactly `AC_SEARCH_LIBS` patched, with `-l$lib` replaced by `$lib.lib`.
I submitted a patch to make `AC_SEARCH_LIBS` try the MSVC syntax if `-l$lib` failed, but it was rejected.

> That sounds awkward for everyone else, no? It slows down 'configure' and might run afoul of ".lib" files present for some other purpose.
> Can you work around the problem by writing a script wrapper around the compiler in question? 

[[PATCH] AC_SEARCH_LIBS now searches for .lib files](https://lists.gnu.org/archive/html/autoconf-patches/2025-01/msg00004.html)

I'll apply your smoke screen.

---

Hmm, given that presumably at some point the detection of the C compiler knows that it's `cl`, I wonder if the `AC_SEARCH_LIBS` machinery could be revisited to select between assuming `-lfoo` or `foo.lib` based on probing done in `AC_PROG_CC`, etc. Anyway, for another day...!